src/content-handlers/iiif/modules/uv-contentleftpanel-module/TreeView.ts

```diff
@@ -63,37 +83,34 @@ export class TreeView extends BaseView<ContentLeftPanel> {
   }
 
   public selectNode(node: TreeNode): void {
-    if (!this.treeComponent.selectedNode) {
-      this.treeComponent.expandParents(node, true);
-
-      const link: Element | undefined = this.$tree.find(
-        "#tree-link-" + node.id
-      )[0];
-
-      if (link) {
-        // link.scrollIntoView({ inline: 'center' });
-      }
+    this.treeComponent.expandParents(node, true); // Expand node parents
+    const link: Element | undefined = this.$tree.find("#tree-link-" + node.id)[0];
+    if (link) {
+      //commented out as bug where scrolls to wrong node eg in Villanova collection
+     // link.scrollIntoViewIfNeeded();
     }
 
-    setTimeout(() => {
+    Promise.resolve().then(() => {
       this.treeComponent.selectNode(node);
-    }, 0);
-  }
+  });
+}
 
   public expandNode(node: TreeNode, expanded: boolean): void {
     this.treeComponent.expandNode(node, expanded);
   }
 
   public getAllNodes(): TreeNode[] {
-    return this.treeComponent.getAllNodes();
+    const allNodes = this.treeComponent.getAllNodes();
```

Comment:
Did you refactor this (and the method below) for debugging purposes? If so, should they be put back the way they were now that you're done? (If there's a reason to change a one-liner into a two-liner, I don't really mind -- but otherwise it might be better to remain concise).

Reply:
Tested again today in the Vercel app, and it looks good. The fix has addressed and resolved the issue #1041. Thanks again @Saira-A 👍